### DISCUSSION TITLE
Impressions after running a complete activity (on Linux!)

### DISCUSSION THREAD

**Discussion by HengYeDev** [perpetualKid/FreeTrainSimulator#72] [General]
I just finished running a complete high-speed activity on Linux. (Train D3041, Shanghai-Nanjing Intercity Railway)

Impressions:
* Sometimes weird artifacts show up. When I save and resume the artifacts disappear. Not sure if its a DXVK (linux directx) issue or an open rails issue. Examples: 
![image](https://user-images.githubusercontent.com/62600441/114588889-2a412e00-9c55-11eb-90b5-06f4990f0792.png) 
![image](https://user-images.githubusercontent.com/62600441/114589095-58bf0900-9c55-11eb-89bb-041f32e8abe4.png)
As you can see sometimes its grey and sometimes its red. It is never any other color. It mostly happens in crowded areas.
* AI train behavior is erratic sometimes. They sometimes just disappear after entering stations. This might be an activity problem but I see it in multiple activities.
* Graphics is very fast as expected, get up to 150 FPS.

**Comment by HengYeDev** [comment]
@perpetualKid have you ever seen the artifacting on windows?

**Reply by perpetualKid** [reply]
can't remember to have seen such artifacting, nor have seen any other user reports about :(

**Reply by perpetualKid** [reply]
may be worth though to follow up on the AI train behavior. Any chance you can compare with an original OR version?

**Reply by HengYeDev** [reply]
i can try later today...but i do remember that in a different activity there was a train at a station in regular OR but not in ORMG, at same game time

**Reply by HengYeDev** [reply]
It might also be a activity problem - i'm trying out a 7 hr activity today, i'll see how it turns out

**Comment by HengYeDev** [comment]
@perpetualKid seems like the artifacting only happens in activity mode with AI trains. and the artifacts are attached to AI trains an are the color of the AI train. do you know whether directx or openrails is responsible for loading 3d objects?

**Reply by perpetualKid** [reply]
I think its all loading in OR code, and I wouldn't be much surprised if there is some bug i.e. wroingly indexed/initialized buffers or so. What I'm not quite getting yet is the dependency with AI trains, but I almost assume this is just a red herring and it could be any other train as well, ie. in multiplayer scenario.

We need to isolate and try to make reproducable, so we than could test on Windows, and also compare with other OR versions. Maybe testing on a commonly available route, such as those available on OR website (which seems down at the moment).

**Reply by HengYeDev** [reply]
It never happens on regular OR on same route, same activity, everything the same

**Reply by HengYeDev** [reply]
i've never tried multiplayer because i can't drive two trains at once, but they seem to be 3D Models sticking off the trains, since they behave similar to objects, (e.x. changing perspective and angle displays it differently)

**Reply by HengYeDev** [reply]
Also, it seems that the amount of strange objects/artifacts increases porportionately to session runtime.

**Reply by perpetualKid** [reply]
> It never happens on regular OR on same route, same activity, everything the same

which regular OR have you tested, the Monogame-based, or the old-style XNA?

**Reply by HengYeDev** [reply]
XNA

**Reply by perpetualKid** [reply]
that just adds another level of complexity, there were a few changes required when switching to Monogame, also since OR still was on XNA 3.1 not even 4. 
Let's try to find a reproducable scenario so we can work the other factors

**Reply by HengYeDev** [reply]
ok, i dont have any windows computers though. nothing of value in the DXVK (directx) logs, and cleared the dxvk cache also. so you have maple leaf tracks greater toronto area route?

**Reply by perpetualKid** [reply]
no, I don't have that route, seems it's not a freely available but payware only?
Maybe we can test using any of these [openrails.org](http://www.openrails.org/download/content/) like the "DemoModel 1" or [BNSF Starter Route for OR](https://www.trainsimulations.net/starter)?

**Comment by HengYeDev** [comment]
how many activites do they have and do they have lots of AI traffic?

**Reply by perpetualKid** [reply]
the DemoModel 1 has one activity with some traffic.

If you're active on TrainSim, there is a demo world ["A Small World Signal Test Route"](https://www.trainsim.com/vbts/tsview.php?do=list&fid=7348) which I often used for testing, contains a lot of AI and a bunch of activities. 

_Smaller route with many branch lines, signals and exclusively automatic switches. It includes seven stations (two terminals), two larger yards and several sidings. The length of the route is 50 km, 100 km if you use the loop at one end. Made to test the behaviour of A.I. trains, so enjoy a lot of A.I. traffic on this route!_

Only downside you'd ti hunt a couple more downloads for all the rolling stocks :(

**Reply by HengYeDev** [reply]
the thing is, the artifacts dont come with all the ai traffic, so have to hunt edge cases...

Hunting downloads isnt my strong point...i currently just download a package for all the chinese trains i like to drive from a chinese website (about 5gb, hundreds of rolling stock)

I also dont have msts so i wonder what rolling stock the DomoModel1 uses?

**Reply by HengYeDev** [reply]
btw it would be great if we set up a real time chat for devs/testers so its easier to communicate and we dont have these enormous threads :)

**Reply by perpetualKid** [reply]
> btw it would be great if we set up a real time chat for devs/testers so its easier to communicate and we dont have these enormous threads :)

indeed, I was thinking so myself. Looking into Discord or Gitter, but given integration with Github (Logins) tend to Gitter. Let's give it a try - I've created a channel, and linked from this repo main page  [![Join the chat at https://gitter.im/ORTS-MG/community](https://badges.gitter.im/ORTS-MG/community.svg)](https://gitter.im/ORTS-MG/community?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)

**Comment by HengYeDev** [comment]
yeah gitter is now matrix which i really like



Heng Ye
https://yeheng.org | Twitter https://twitter.com/HengYeDev








---- On Fri, 16 Apr 2021 01:05:17 -0400 perpetualKid ***@***.***> wrote ----




btw it would be great if we set up a real time chat for devs/testers so its easier to communicate and we dont have these enormous threads :)


indeed, I was thinking so myself. Looking into Discord or Gitter, but given integration with Github (Logins) tend to Gitter. Let's give it a try

**Reply by perpetualKid** [reply]
[![Join the chat at https://gitter.im/ORTS-MG/community](https://badges.gitter.im/ORTS-MG/community.svg)](https://gitter.im/ORTS-MG/community?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
